<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>846</width>
    <height>740</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="OpenGLWidget" name="openGLWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="focusPolicy">
           <enum>Qt::StrongFocus</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="horizontalSlider">
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="sliderPosition">
           <number>50</number>
          </property>
          <property name="tracking">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="invertedAppearance">
           <bool>false</bool>
          </property>
          <property name="invertedControls">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox">
          <item>
           <property name="text">
            <string>Original</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Edge Detection</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Gaussian Blur</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Bloom</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Gamma Correction</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Toon</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Fish Eye</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Billboard</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="settingsLayout">
          <item>
           <layout class="QVBoxLayout" name="labelsLayout"/>
          </item>
          <item>
           <layout class="QVBoxLayout" name="slidersLayout"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="bottomLayout">
          <item>
           <widget class="QPushButton" name="buttonOpen">
            <property name="text">
             <string>Open</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonSave">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonExit">
            <property name="text">
             <string>Exit</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>846</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>openglwidget.h</header>
   <slots>
    <slot>slotSendImage()</slot>
    <slot>shaderChanged(int)</slot>
    <slot>sliderChanged(int)</slot>
    <slot>setEdgeThreshold(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>slotOpenFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonOpen</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slotOpenFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>openGLWidget</receiver>
   <slot>slotSendImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonSave</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>slotSendImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>211</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>shaderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonExit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>698</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>sliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>618</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>shaderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>652</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotOpenFile()</slot>
  <slot>shaderChanged(int)</slot>
 </slots>
</ui>
